pkg/client/client.go

```diff
@@ -325,7 +328,26 @@ func New(c *restclient.Config) (*Client, error) {
 		return nil, err
 	}
 
-	return &Client{client}, nil
+	return &Client{RESTClient: client, config: &config}, nil
+}
+
+// ForGroupVersion returns a copy of the client but without any openshift legacy
```

Comment:
In fact, for everything new this is a very good idea 👍  There is no sense to produce legacy code if we can avoid it.